diff --git a/src/specfem2D/check_grid.F90 b/src/specfem2D/check_grid.F90
index 34182543..ec54898f 100644
--- a/src/specfem2D/check_grid.F90
+++ b/src/specfem2D/check_grid.F90
@@ -480,9 +480,19 @@
 
 
   ! creates a PostScript file with stability condition
+  !if (output_postscript_snapshot) then
+  !  call check_grid_create_postscript(courant_stability_number_max,lambdaPImin,lambdaPImax,lambdaSmin,lambdaSmax)
+  !endif
+
+  !----------------------------------------------------------------!
+  !Modified: forcing output postscript of grids (Kurama 05/06/2019)
+  !----------------------------------------------------------------!
+  output_postscript_snapshot = .true.
   if (output_postscript_snapshot) then
     call check_grid_create_postscript(courant_stability_number_max,lambdaPImin,lambdaPImax,lambdaSmin,lambdaSmax)
   endif
+  output_postscript_snapshot = .false.
+  !----------------------------------------------------------------!
 
   end subroutine check_grid
 
